proposals/lip-0055.md

```diff
@@ -300,25 +358,149 @@ blockHeaderSchema = {
+  "type": "object",
+  "required": [
+    "version",
+    "timestamp",
+    "height",
+    "previousBlockID",
+    "generatorAddress",
+    "transactionRoot",
+    "assetRoot",
+    "eventRoot",
+    "stateRoot",
+    "maxHeightPrevoted",
+    "maxHeightGenerated",
+    "validatorsHash",
+    "aggregateCommit"
+  ],
+  "properties": {
+    "version": {
+      "dataType": "uint32",
+      "fieldNumber": 1
+    },
+    "timestamp": {
+      "dataType": "uint32",
+      "fieldNumber": 2
+    },
+    "height": {
+      "dataType": "uint32",
+      "fieldNumber": 3
+    },
+    "previousBlockID": {
+      "dataType": "bytes",
+      "fieldNumber": 4
+    },
+    "generatorAddress": {
+      "dataType": "bytes",
+      "fieldNumber": 5
+    },
+    "transactionRoot": {
+      "dataType": "bytes",
+      "fieldNumber": 6
+    },
+    "assetRoot": {
+      "dataType": "bytes",
+      "fieldNumber": 7
+    },
+    "eventRoot": {
+      "dataType": "bytes",
+      "fieldNumber": 8
+    },
+    "stateRoot": {
+      "dataType": "bytes",
+      "fieldNumber": 9
+    },
+    "maxHeightPrevoted": {
+      "dataType": "uint32",
+      "fieldNumber": 10
+    },
+    "maxHeightGenerated": {
+      "dataType": "uint32",
+      "fieldNumber": 11
+    },
+    "validatorsHash": {
+      "dataType": "bytes",
+      "fieldNumber": 12
+    },
+    "aggregateCommit": {
+      "type": "object",
+      "fieldNumber": 13,
+      "required": [
+        "height",
+        "aggregationBits",
+        "certificateSignature"
+      ],
+      "properties": {
+        "height": {
+          "dataType": "uint32",
+          "fieldNumber": 1
+        },
+        "aggregationBits": {
+          "dataType": "bytes",
+          "fieldNumber": 2
+        },
+        "certificateSignature": {
+          "dataType": "bytes",
+          "fieldNumber": 3
+        }
+      }
+    }
+  }
+}
+```
+
+
+#### Block Signature Calculation
+
+The block header signature is calculated using the `calculateBlockSignature` function. This function returns a bytes value with length `SIGNATURE_LENGTH_EDDSA_BYTES`.
+
+```python
+calculateBlockSignature():
+  let secretKey be the secret generator key of the block generator
+  # Serialize the block header without signature
+  let serializedUnsignedBlockHeader be the serialization of block.header following the unsignedBlockHeaderSchema
+  let networkIdentifier be the network identifier of the chain
+
+  return signMessage(secretKey, "LSK_BH_", networkIdentifier, serializedUnsignedBlockHeader)
+```
+
+Here, the function `signMessage` calculates the signature of a message as specified in [LIP 0037][lip-0037#signing-and-verifying-with-ed25519].
+
+#### Block Header Generation
+
+During block generation, the block header properties are set in the following stages.
+
+- **Header initialization**:
+  - Set the `version`, `timestamp`, `height`, `previousBlockID`, `generatorAddress`, `maxHeightPrevoted`, `maxHeightGenerated` properties.
+- **Header finalization**:
```

Comment:
I think the `aggregateCommit` should be moved to the header initialization as it is also verified with respect to the state before applying the state changes implied by the block.

@shuse2 Are all the other properties in the same stage in the implementation?

Reply:
`AggregateCommit` was actually inserted in the finalization step in the implementation now. I will create an issue to fix this.
Otherwise, the list is aligned with the current implementation

Reply:
fixed